src/WebJobs.Script/Diagnostics/DictionaryLoggerScope.cs

```diff
@@ -13,27 +14,56 @@ namespace Microsoft.Azure.WebJobs.Script.Diagnostics
     /// </summary>
     internal class DictionaryLoggerScope
     {
-        private readonly object _state;
-
         private static AsyncLocal<DictionaryLoggerScope> _value = new AsyncLocal<DictionaryLoggerScope>();
 
-        private DictionaryLoggerScope(object state, DictionaryLoggerScope parent)
+        private DictionaryLoggerScope(IReadOnlyDictionary<string, object> state, DictionaryLoggerScope parent)
         {
-            _state = state;
+            State = state;
             Parent = parent;
         }
 
+        internal IReadOnlyDictionary<string, object> State { get; private set; }
+
         internal DictionaryLoggerScope Parent { get; private set; }
 
         public static DictionaryLoggerScope Current
         {
-            get => _value.Value;
-            set => _value.Value = value;
+            get
+            {
+                return _value.Value;
+            }
+
+            set
+            {
+                _value.Value = value;
+            }
         }
 
         public static IDisposable Push(object state)
         {
-            Current = new DictionaryLoggerScope(state, Current);
+            var stateValues = state as IDictionary<string, object>;
+            if (stateValues == null)
+            {
+                var stateEnum = state as IEnumerable<KeyValuePair<string, object>>;
```

Comment:
I copied this from the SDK and this is one of those "StyleCop forced me to use C#6" examples. Now I can update it!